### PR TITLE
aws ecr get-login is deprecated since v1. Using  get-login-password instead, issue 3013

### PR DESCRIPTION
**Pull request by mrugeles**
## What changes are proposed in this pull request?

(Please fill in changes proposed in this fix)

## How is this patch tested?

(Details)

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for
Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [x] `integrations/sagemaker`: SageMaker integrations

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [x] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by dbczumar — CHANGES_REQUESTED** [review]
@mrugeles It looks like `aws ecr get-login-password` isn't a drop-in replacement for `get-login`. `get-login-password` returns a password that must then be submitted to the `docker login` command, as described here: https://docs.aws.amazon.com/cli/latest/reference/ecr/get-login-password.html#examples

For example:

```
aws ecr get-login-password \
    --region <region> \
| docker login \
    --username AWS \
    --password-stdin <aws_account_id>.dkr.ecr.<region>.amazonaws.com
```

Can you make this change to pass the generated password to `docker login`, as above? Let me know if you have any questions!

**Comment by mrugeles** [comment]
> https://docs.aws.amazon.com/cli/latest/reference/ecr/get-login-password.html#examples

What would be the best way to retrieve  `<aws_account_id>` and `<region>`? Via AWS CLI command?

**Comment by dbczumar** [comment]
@mrugeles I think those are already available in the same function; they're fetched here: https://github.com/mlflow/mlflow/blob/7c3c06df6104ecc4e180bece67d39c42097556e1/mlflow/sagemaker/__init__.py#L113-L115

**Comment by mrugeles** [comment]
> @mrugeles I think those are already available in the same function; they're fetched here:
> 
> https://github.com/mlflow/mlflow/blob/7c3c06df6104ecc4e180bece67d39c42097556e1/mlflow/sagemaker/__init__.py#L113-L115

I miss that one, thx.

**Comment by mrugeles** [comment]
The command was tested ok for Windows and MacOs

File: mlflow/sagemaker/__init__.py [hunk 1]
```diff
@@ -126,16 +126,15 @@ def push_image_to_ecr(image=DEFAULT_IMAGE_NAME):
...
+    docker_login_cmd = "aws ecr get-login-password" \
+                       " | docker login  --username AWS " \
+                       " --password-stdin " \
+                       "{account}.dkr.ecr.{region}.amazonaws.com".format(
+                            account=account,
+                            region=region)
+
     os_command_separator = ";\n"
-    if platform.system() == "Windows":
-        os_command_separator = " && "
```
Comment (dbczumar) [inline mlflow/sagemaker/__init__.py:131]: I think we still need the ` && ` OS command separator for windows, right?

File: mlflow/sagemaker/__init__.py [hunk 2]
```diff
@@ -126,16 +126,15 @@ def push_image_to_ecr(image=DEFAULT_IMAGE_NAME):
     # x = ecr_client.get_authorization_token()['authorizationData'][0]
     # docker_login_cmd = "docker login -u AWS -p {token} {url}".format(token=x['authorizationToken']
     #                                                                ,url=x['proxyEndpoint'])
+
+    docker_login_cmd = "aws ecr get-login-password" \
```
Comment (dbczumar) [inline mlflow/sagemaker/__init__.py:130]: Was this tested in a windows command prompt or the windows linux subsystem?

Comment (mrugeles) [reply mlflow/sagemaker/__init__.py:130]: It was tested from the windows command prompt

File: mlflow/sagemaker/__init__.py (hunk 1 above)
Comment (mrugeles) [reply mlflow/sagemaker/__init__.py:131]: Yes, you're right, I'll add it back.

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @mrugeles

**Comment by dbczumar** [comment]
Hi @mrugeles ,  since MLflow has joined the Linux Foundation, we now require each PR to be signed using git commit -s. Can you follow the instructions here so that the DCO tests will pass?: https://github.com/mlflow/mlflow/blob/master/CONTRIBUTING.rst#sign-your-work

**Comment by mrugeles** [comment]
> Hi @mrugeles , since MLflow has joined the Linux Foundation, we now require each PR to be signed using git commit -s. Can you follow the instructions here so that the DCO tests will pass?: https://github.com/mlflow/mlflow/blob/master/CONTRIBUTING.rst#sign-your-work

Done :)

**Pull request merged**